fuzzamoto-scenarios/bin/ir.rs

```diff
@@ -200,17 +213,35 @@ where
                     if self.inner.connections.is_empty() {
                         return;
                     }
-
                     let num_connections = self.inner.connections.len();
-                    if let Some(connection) = self.inner.connections.get_mut(from % num_connections)
-                    {
+                    let dst = from % num_connections;
+
+                    if let Some(connection) = self.inner.connections.get_mut(dst) {
                         if cfg!(feature = "force_send_and_ping") {
-                            let _ = connection.send_and_ping(&(command, message));
+                            if let Ok(received) = connection.send_and_recv(
+                                &(command, message),
+                                self.recording_received_messages,
+                            ) {
+                                if !received.is_empty() {
+                                    // don't add empty vev
+                                    self.received.extend(
+                                        received.into_iter().map(|(s, bytes)| (dst, s, bytes)),
+                                    );
+                                }
```

Comment:
nit: i don't think we need the `is_empty` check, calling `extend` with an empty vector should be nop?